projects/storefrontlib/src/cms-components/checkout/components/place-order/place-order.component.ts

```diff
@@ -96,7 +117,15 @@ export class PlaceOrderComponent implements OnInit, OnDestroy {
     this.subscription.add(
       this.checkoutReplenishmentFormService
         .getScheduleReplenishmentFormData()
-        .subscribe((data) => (this.scheduleReplenishmentFormData = data))
+        .subscribe((data) => {
+          this.scheduleReplenishmentFormData = data;
+
+          this.daysOfWeekNotChecked =
+            data.daysOfWeek.length === 0 &&
+            data.recurrencePeriod === recurrencePeriod.WEEKLY;
+
+          this.cd.markForCheck();
```

Comment:
If you make `daysOfWeekNotChecked` a BehaviorSubject and expose it to the template as an Observable you can observe it with the async pipe and remove the ChangeDetectorRef.